src/test/java/DemoScript.java

```diff
@@ -0,0 +1,128 @@
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.chrome.ChromeOptions;
+import org.openqa.selenium.edge.EdgeDriver;
+import org.openqa.selenium.edge.EdgeOptions;
+import org.openqa.selenium.firefox.FirefoxDriver;
+import org.openqa.selenium.firefox.FirefoxOptions;
+import org.openqa.selenium.safari.SafariDriver;
+import org.openqa.selenium.safari.SafariOptions;
+import org.testng.ITestContext;
+import org.testng.SkipException;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.AfterTest;
+import org.testng.annotations.Parameters;
+import org.testng.annotations.Test;
+import org.openqa.selenium.Dimension;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+public class DemoScript {
+    public WebDriver driver;
+    public String browser;
+    public String url;
+    protected String PROJECT_TOKEN;
+    protected Dimension[] resolutions;
+
+    @BeforeTest
+    @Parameters({"browser", "resolutions", "url"})
+    public void setup(String browser, String resolutionsCSV, String url) {
+        this.url = url;
+        this.browser = browser;
+        Map<String, String> envMap = System.getenv();
+        this.PROJECT_TOKEN = envMap.getOrDefault("PROJECT_TOKEN", "");
+        this.resolutions = Arrays.stream(resolutionsCSV.split(",")).map((resolution) ->new Dimension(Integer.valueOf(resolution.split("x")[0]), Integer.valueOf(resolution.split("x")[1]))).toArray(Dimension[]::new);
+        driver = setupBrowserDriver(browser, driver);
+    }
+
+    @AfterTest
+    public void tearDown() {
+        driver.close();
+    }
+
+    @Test
+    public void DemoScript(ITestContext ctx) throws Exception {
+        String suiteName = ctx.getCurrentXmlTest().getSuite().getName();
+        VisualTest visualTest = new VisualTest(driver, buildSettings(suiteName, browser));
+        for (Dimension resolution : this.resolutions) {
+            driver.manage().window().setSize(resolution);
+            driver.get(this.url);
+            Thread.sleep(2000); //some pages have a slight loading time
+            String imageName = String.format("DemoProject-%s-%s", browser, resolution);
+            visualTest.capture(imageName);
+        }
+    }
+
+    public HashMap<String, Object> buildSettings(String suiteName, String browser) {
+        HashMap<String, Object> settings = new HashMap<>();
+        settings.put("testRunName", suiteName + ": " + browser);
+        settings.put("debug", false);
+        settings.put("projectToken", this.PROJECT_TOKEN);
+
+        Path currentPath = Paths.get(System.getProperty("user.dir"));
+        Path filePath = Paths.get(currentPath.toString(), String.format("src/test/resources/screenshots/%s", suiteName));
+        settings.put("saveTo", filePath.toString());
+        return settings;
+    }
+
+    public static WebDriver setupBrowserDriver(String browser, WebDriver driver){
```

Comment:
I don't remember if Safari is working or not. But You can get rid of it if it's not working. Chrome and Firefox would be fine.

Reply:
Safari is working for current sdk version.